### PR TITLE
Improve the master/main renaming guide

### PR DESCRIPTION
**Pull request by premun**
- Add more screenshots
- Improve prerequisite instructions
- Shorten step names
- Make overview ToC clickable
- Add section about Component Governance
- Other minor clarity improvements

### CODE REVIEW

File: Documentation/Projects/M2MRenaming/Master-to-Main-renaming-guide.md [hunk 1]
```diff
@@ -165,58 +179,71 @@ pr:
...
-  > .\list-repo-pipelines.ps1 -GitHubRepository "[Your repo in the form dotnet/runtime]" -PAT "[personal access token]"
+  > .\list-repo-pipelines.ps1 -GitHubRepository "[REPOSITORY]" -PAT "[TOKEN]"
   > ```
 
+Example:
+  > ```ps
+  > .\list-repo-pipelines.ps1 -GitHubRepository "dotnet/runtime" -PAT "jdsvmd324jnsdvjafn2vsd"
+  > ```
+
+The **Pesonal Access Token** needs following scopes: {TODO}.
```
Comment (jakubstilec) [inline Documentation/Projects/M2MRenaming/Master-to-Main-renaming-guide.md:214]: Just to make sure - is {TODO} forgotten or you want to add it later?

**Review by jakubstilec — APPROVED** [review]

File: Documentation/Projects/M2MRenaming/Master-to-Main-renaming-guide.md (hunk 1 above)
Comment (premun) [reply Documentation/Projects/M2MRenaming/Master-to-Main-renaming-guide.md:214]: Good catch, I'll figure that now

File: Documentation/Projects/M2MRenaming/Master-to-Main-renaming-guide.md [hunk 2]
```diff
@@ -165,58 +179,71 @@ pr:
...
+5. Save the changes
+   ![Pipeline triggers](images/save-pipeline.png)
+
 
+## 9. Switch the default branch of the AzDO repository to `main`
 
-## 9. ![AzDO mirrored](images/azdo-mirrored.png) Go to the [internal AzDO dnceng](https://dev.azure.com/dnceng/internal/_git) mirror of your repository and switch the default branch to `main`
+![AzDO mirrored](images/azdo-mirrored.png) 
 
+Go to the [internal AzDO dnceng](https://dev.azure.com/dnceng/internal/_git) mirror of your repository and switch the default branch to `main`:
```
Comment (tkapin) [inline Documentation/Projects/M2MRenaming/Master-to-Main-renaming-guide.md:231]: The picture is nice, but the exact instructions would further clarify this.

**Review by tkapin — CHANGES_REQUESTED** [review]

File: Documentation/Projects/M2MRenaming/Master-to-Main-renaming-guide.md (hunk 1 above)
Comment (premun) [reply Documentation/Projects/M2MRenaming/Master-to-Main-renaming-guide.md:214]: Yup, I was gonna tackle the PS script next

Comment (premun) [reply Documentation/Projects/M2MRenaming/Master-to-Main-renaming-guide.md]: I will take care of this together with the PS script error handling in a separate PR. I put in my best guess.

**Review by tkapin — APPROVED** [review]

**Pull request merged**